src/types/Types.tsx

```diff
@@ -0,0 +1,302 @@
+  | { Inactive: { someValue: number } }
+  | { Abandoned: {} };
+
+export type TransactionInfoType = {
+  txid: string;
+  tx_type: TxType;
+  timestamp: { timestamp: number };
+  state: TxState;
+};
+
+export type TransactionType = {
+  count: number;
+  skip: number;
+  total: number;
+  txs: TransactionInfoType[];
+};
+
+export type WalletInfo = {
+  wallet_id: number;
+  path: string;
+  encryption: string;
+  accounts: Record<string, AccountType>;
+  best_block: [string, number];
+  wallet_type: string;
+};
+
+export type AddressInfo = {
+  wallet_id: number;
+  account_id: string;
+  index: number;
+  address: string;
+};
+
+export type NewAccountResultType = {
+  wallet_id: string;
+  account_id: string;
+  account_info: AccountType;
+};
+
+export type ToggleStakingResultType = {
+  wallet_id: string;
+  account_id: string;
+  enabled: boolean;
+};
+
+export type ChainInfoType = {
+  best_block_height: number;
+  best_block_id: string;
+  best_block_timestamp: {
+    timestamp: number;
+  };
+  median_time: {
+    timestamp: number;
+  };
+  is_initial_block_download: boolean;
+};
+
+export type InitNodeType = {
+  chain_info: ChainInfoType;
+  empty_consensus_reward_maturity_block_count: number;
+};
+
+export type PeerDisconnected = {
+  P2p: {
+    PeerDisConnected: number;
+  };
+};
+
+export type PeerConnected = {
+  PeerConnected: {
+    id: number;
+    services: number;
+    address: string;
+    inbound: boolean;
+    user_agent: number[];
+    software_version: {
+      major: number;
+      minor: number;
+      patch: number;
+    };
+  };
+};
+
+export type P2p = PeerConnected | PeerDisconnected;
+
+export type Transaction = {
+  V1: {
+    version: number | null; // Assuming version can be a number or null
+    flags: number; // Assuming flags is a number
+    inputs: Input[]; // Array of Input objects
+    outputs: Output[]; // Array of Output objects
+  };
+};
+
+export type Input = Utxo | Account;
+export type Account = {
+  Account: {
+    account: {
+      DelegationBalance: [string, { atoms: string }];
+    };
+    nonce: number;
+  };
+};
+
+export type Utxo = {
+  Utxo: {
+    id: {
+      Transaction: string; // Transaction ID as a string
+    };
+    index: number; // Index as a number
+  };
+};
+
+export type Output =
+  | CreateStakePoolOutput
+  | TransferOutput
+  | LockThenTransferOutput
+  | DelegateStakingOutput
+  | IssueFungibleTokenOutput
+  | ProduceBlockFromStakeOutput
+  | CreateDelegationId;
+
+export type CreateStakePoolOutput = {
+  CreateStakePool: [
+    string,
+    {
+      pledge: {
+        atoms: string;
+      };
+      staker: string;
+      vrf_public_key: string;
+      decomission_key: string;
+      margin_ratio_per_thousand: string;
+      cost_per_block: {
+        atoms: string;
+      };
+    }
+  ];
+};
+
+export type CreateDelegationId = {
+  CreateDelegationId: [string, string];
+};
+
+export type TransferOutput = {
+  Transfer: [
+    {
+      Coin: {
+        atoms: string;
```

Comment:
1. Transfer and LockThenTransfer output type can contain not only Coins but also a Token.
2. use decimals instead of atoms to avoid manual calculations and errors later